### PR TITLE
Feature: Add test for TmGetPeerShardInfoV2

### PR DESCRIPTION
**Pull request by lwojtow**

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -105,12 +105,23 @@ The test index makes use of symbolic language in describing connection and messa
 
 ### ZG-CONFORMANCE-004
     
-    The node responds with mtLEDGER_DATA for mtGET_LEDGER.
+    The node responds with mtLEDGER_DATA for mtGET_LEDGER with iType == LiBase.
```
Comment (niklaslong) [inline SPEC.md:108]: It might be worth concisely documenting what `iType` and `LiBase` mean here?

File: SPEC.md [hunk 2]
```diff
@@ -105,12 +105,23 @@ The test index makes use of symbolic language in describing connection and messa
...
     
     <>
     -> mtGET_LEDGER with iType == LiBase
     <- mtLEDGER_DATA
-    or:
 
+### ZG-CONFORMANCE-005
+
+    The node responds with mtLEDGER_DATA for mtGET_LEDGER with iType == LiAsNode.
```
Comment (niklaslong) [inline SPEC.md:116]: ditto

File: SPEC.md (hunk 1 above)
Comment (lwojtow) [reply SPEC.md:108]: As per slack: agreed to elaborate on this in the near future.

File: SPEC.md [hunk 3]
```diff
@@ -105,12 +105,23 @@ The test index makes use of symbolic language in describing connection and messa
...
+
+    <>
     -> mtGET_LEDGER with iType == LiAsNode
     <- mtLEDGER_DATA
+
+### ZG-CONFORMANCE-006
+
+    The node requests mtGET_PEER_SHARD_INFO_V2 after start up.
+
+    ->
```
Comment (niklaslong) [inline SPEC.md:126]: Does this include the handshake or is that message sent immediately?

Comment (lwojtow) [reply SPEC.md:126]: Looking at the code it seems like it's after the handshake.

Comment (niklaslong) [reply SPEC.md:126]: We might want to find a way to make that clear in the symbolic language if we ever have cases that don't include the handshake, for now it's fine 👍

**Review by niklaslong — APPROVED** [review]

**Pull request merged**